fxa-client/src/http_client/mod.rs

```diff
@@ -255,23 +261,117 @@ impl<'a> Client<'a> {
             .header(header::CONTENT_TYPE, "application/json")
             .body(body.to_string())
             .build()?;
-        Client::make_request(request)?;
+        Self::make_request(request)?;
+        Ok(())
+    }
+
+    pub fn pending_commands(&self, refresh_token: &str) -> Result<PendingCommandsResponse> {
+        let url = self
+            .config
+            .auth_url_path("v1/client_instance/pending_commands")?;
+        let client = ReqwestClient::new();
+        let request = client
+            .request(Method::GET, url)
+            .header(header::AUTHORIZATION, Self::bearer_token(refresh_token))
+            .build()?;
+        Self::make_request(request)?.json().map_err(|e| e.into())
+    }
+
+    pub fn invoke_command(
+        &self,
+        access_token: &str,
+        command: &str,
+        target: &str,
+        payload: &serde_json::Value,
+    ) -> Result<()> {
+        let body = json!({
+            "command": command,
+            "target": target,
+            "payload": payload
+        });
+        let url = self
+            .config
+            .auth_url_path("v1/clients_instances/invoke_command")?;
+        let client = ReqwestClient::new();
+        let request = client
+            .request(Method::POST, url)
+            .header(header::AUTHORIZATION, Self::bearer_token(access_token))
+            .header(header::CONTENT_TYPE, "application/json")
+            .body(body.to_string())
+            .build()?;
+        Self::make_request(request)?;
+        Ok(())
+    }
+
+    pub fn client_instance(&self, refresh_token: &str) -> Result<ClientInstanceResponse> {
+        let url = self.config.auth_url_path("v1/client_instance")?;
+        let client = ReqwestClient::new();
+        let request = client
+            .request(Method::GET, url)
+            .header(header::AUTHORIZATION, Self::bearer_token(refresh_token))
+            .build()?;
+        Self::make_request(request)?.json().map_err(|e| e.into())
+    }
+
+    pub fn clients_instances(&self, access_token: &str) -> Result<Vec<ClientInstanceResponse>> {
```

Comment:
Naming nit: something about double plural "clients instances" here just doesn't seem right to me; can this be something like "client instances"?